src/lightning_lite/strategies/strategy.py

```diff
@@ -0,0 +1,300 @@
+        """Perform an all_gather on all processes.
+
+        Args:
+            tensor: the tensor to all_gather
+            group: the process group to gather results from
+            sync_grads: flag that allows users to synchronize gradients for all_gather op
+        """
+
+    def reduce_boolean_decision(self, decision: bool) -> bool:
+        """Reduce a boolean decision across all processes."""
+        return decision
+
+    def save_checkpoint(
+        self, checkpoint: Dict[str, Any], filepath: _PATH, storage_options: Optional[Any] = None
+    ) -> None:
+        """Save model/training states as a checkpoint file through state-dump and file-write.
+
+        Args:
+            checkpoint: dict containing model and trainer state
+            filepath: write-target file's path
+            storage_options: parameter for how to save to storage, passed to ``CheckpointIO`` plugin
+        """
+        if self.is_global_zero:
+            self.checkpoint_io.save_checkpoint(checkpoint, filepath, storage_options=storage_options)
+
+    def get_module_state_dict(self, module: Module) -> Dict[str, Union[Any, Tensor]]:
+        """Returns model state."""
+        return module.state_dict()
+
+    def get_optimizer_state(self, optimizer: Optimizer) -> Dict[str, Tensor]:
+        """Returns state of an optimizer.
+
+        Allows for syncing/collating optimizer state from processes in custom plugins.
+        """
+        return optimizer.state_dict()
+
+    def load_checkpoint(self, checkpoint_path: _PATH) -> Dict[str, Any]:
+        torch.cuda.empty_cache()
+        return self.checkpoint_io.load_checkpoint(checkpoint_path)
+
+    def load_module_state_dict(self, module: Module, checkpoint: Mapping[str, Any]) -> None:
+        module.load_state_dict(checkpoint["state_dict"])
+
+    def load_optimizer_state_dict(
+        self, optimizers: Union[Optimizer, Iterable[Optimizer]], checkpoint: Mapping[str, Any]
+    ) -> None:
+        if not isinstance(optimizers, Iterable):
+            optimizers = [optimizers]
+        optimizer_states = checkpoint["optimizer_states"]
+        for optimizer, opt_state in zip(optimizers, optimizer_states):
+            optimizer.load_state_dict(opt_state)
+            optimizer_to_device(optimizer, self.root_device)
+
+    def remove_checkpoint(self, filepath: _PATH) -> None:
+        """Remove checkpoint filepath from the filesystem.
+
+        Args:
+            filepath: Path to checkpoint
+        """
+        if self.is_global_zero:
+            self.checkpoint_io.remove_checkpoint(filepath)
+
+    def teardown(self) -> None:
+        """This method is called to teardown the training process.
+
+        It is the right place to release memory and free other resources.
+        """
+        self.precision_plugin.teardown()
+        assert self.accelerator is not None
+        self.accelerator.teardown()
+        self.checkpoint_io.teardown()
+
```

Comment:
Reviewer: For Lite, there is no teardown that happens automatically. I would suggest to either
A) Remove it
B) Provide a method specific to freeing memory (that the user could call via lite.free_memory()) for example.

Reply:
I would call for C) but dot have any strong preference here :otter:

Reply:
Can't we call it automatically after `run` is finished?

Reply:
@justusschock We could but I guess what I'm trying to say here is that teardown() is a broad term and in PL it does a lot of things, while in Lite it will never be able to conceptually do the same as in PL. Some example: Moving the LightningModule to CPU, unwrapping the module's sync-batchnorm layers, etc. For this reason, I was hoping in Lite we could make the APIs much more focused, and then in PL we could still keep the very general purpose teardown.

Reply:
In PL we have setup and teardown, so in Lite we could have setup_device and teardown_device, which only does the device-related stuff and could be called automatically. This could then be extended in PL to do the more general purpose things

Reply:
Yes exactly, that's what I mean by focused functions. So we are aligned. 

Lite doesn't do this today, which is why I'm in favor of A/B until we will actually add it to Lite's workflow. As we are currently focused on just moving things to the package in this first iteration.

I'm creating an issue for it so we can work on it.